### PR TITLE
Expenses_Incomes_Basic_Forms

### PR DESCRIPTION
**Pull request by AnastasiiaZah**
- Added 'New' form for Expenses and Incomes. 
- Rubocop file also was renamed.

### CODE REVIEW

File: app/controllers/expenses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class ExpensesController < ApplicationController
+  def index
+    @expense = Expense.all
```
Comment (sleaz0id) [inline app/controllers/expenses_controller.rb]: We shouldn't initialize redundant actions. There is no place in your app where you need `expanses#index` now :).

File: app/controllers/expenses_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  def new
+    @expense = Expense.new
+  end
+
+  def create
+    @expense = Expense.new(expense_params)
+    @expense.build_balance(value: 1)
```
Comment (sleaz0id) [inline app/controllers/expenses_controller.rb]: Value shouldn't be hardcoded - you need to pass this through params :)

File: app/controllers/incomes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class IncomesController < ApplicationController
+  def index
```
Comment (sleaz0id) [inline app/controllers/incomes_controller.rb]: As above :)

File: app/controllers/incomes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      end
+    end
+  end
+
+  private
+
+  def income_params
+    params.require(:income).permit(:title, :value, :category_id)
+  end
+
```
Comment (sleaz0id) [inline app/controllers/incomes_controller.rb]: Redundant empty line :).

File: config/routes.rb [hunk 5]
```diff
@@ -1,4 +1,7 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   root to: 'pages#home'
+
+  resources :expenses
```
Comment (sleaz0id) [inline config/routes.rb]: We can use `only: [:show, :new]` for now :).

File: Gemfile [hunk 6]
```diff
@@ -61,3 +60,4 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+gem 'chartkick'
```
Comment (jerryskye) [inline Gemfile]: Please add a newline character at the end of file :wink:

File: app/views/expenses/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    <%= form.label :category_id %>
+    <%= form.collection_select(:category_id, Category.all, :id, :title) %>
+  </div>
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
+
+
+<% end %>
```
Comment (jerryskye) [inline app/views/expenses/new.html.erb]: Same here, add newline 😜

File: app/views/expenses/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  <div class="field">
+    <%= form.label :category_id %>
+    <%= form.collection_select(:category_id, Category.all, :id, :title) %>
+  </div>
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
+
+
```
Comment (jerryskye) [inline app/views/expenses/new.html.erb]: Redundant empty lines :blush:

File: app/views/incomes/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    <%= form.label :category_id %>
+    <%= form.collection_select(:category_id, Category.all, :id, :title) %>
+  </div>
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
+
+
+<% end %>
```
Comment (jerryskye) [inline app/views/incomes/new.html.erb]: Add newline character 🙂

File: app/views/incomes/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  <div class="field">
+    <%= form.label :category_id %>
+    <%= form.collection_select(:category_id, Category.all, :id, :title) %>
+  </div>
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
+
+
```
Comment (jerryskye) [inline app/views/incomes/new.html.erb]: Same here, remove empty lines :wink:

File: app/controllers/expenses_controller.rb (hunk 2 above)
Comment (AnastasiiaZah) [reply app/controllers/expenses_controller.rb]: Handle it later

File: app/assets/javascripts/application.js [hunk 11]
```diff
@@ -13,4 +13,7 @@
 //= require rails-ujs
 //= require activestorage
 //= require turbolinks
+//= require chartkick
+//= require Chart.bundle
 //= require_tree .
+
```
Comment (sleaz0id) [inline app/assets/javascripts/application.js]: Do we need this empty line?

File: app/views/incomes/new.html.erb [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+  <div class="field">
+    <%= form.label :category_id %>
+    <%= form.collection_select(:category_id, Category.all, :id, :title) %>
+  </div>
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
+
+<%= link_to 'Back', root_url %>
```
Comment (sleaz0id) [inline app/views/incomes/new.html.erb]:
```suggestion
  <%= link_to 'Back', root_url %>
```

File: app/views/pages/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,24 @@
+<h2> incomes </h2>
+
```
Comment (sleaz0id) [inline app/views/pages/index.html.erb]: Redundant empty line

File: app/views/pages/index.html.erb [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+ <table>
+ <tbody>
```
Comment (sleaz0id) [inline app/views/pages/index.html.erb]: The whole block should be indented :).

File: config/initializers/chartkick.rb [hunk 15]
```diff
@@ -0,0 +1,4 @@
+Chartkick.options = {
+    height: "300px",
```
Comment (sleaz0id) [inline config/initializers/chartkick.rb]: We should use 2 spaces for indentation :).

File: db/seeds.rb [hunk 16]
```diff
@@ -5,3 +5,4 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
```
Comment (sleaz0id) [inline db/seeds.rb]: Nope ;)

File: app/controllers/expenses_controller.rb [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+      if @expense.save
+        format.html { redirect_to root_path, notice: 'Expense was successfully created.' }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  private
+  def expense_params
```
Comment (jerryskye) [inline app/controllers/expenses_controller.rb]: An empty line after `private` wouldn't hurt 😜

File: app/assets/javascripts/application.js [hunk 18]
```diff
@@ -13,4 +13,6 @@
 //= require rails-ujs
 //= require activestorage
 //= require turbolinks
-//= require_tree .
+//= require chartkick
+//= require Chart.bundle
+//= require_tree .
```
Comment (jerryskye) [inline app/assets/javascripts/application.js]: The newline character is missing again, I suggest changing your editor's settings or something to detect them 🤔

File: db/seeds.rb [hunk 19]
```diff
@@ -4,4 +4,4 @@
 # Examples:
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+#   Character.create(name: 'Luke', movie: movies.first)
```
Comment (jerryskye) [inline db/seeds.rb]: Here as well :grin:

**Pull request merged**